tests/test_esm_runscripts/test_namelist.py

```diff
@@ -41,3 +60,24 @@ def test_nmls_load(self):
         config = namelists.Namelist.nmls_load(self.mconfig)
         print(config)
         assert config
+
+    def test_nmls_syntax(self):
+        """Checks that the syntax errors are catched and reported to the user"""
+        # Break the namelist by removing the last line
+        with open(self._thisrun_namelist_path, "r") as f:
+            lines = f.readlines()
+        with open(self._thisrun_namelist_path, "w") as f:
+            for line in lines[:-1]:
+                f.write(line)
+        # Captures output (i.e. the user-friendly error)
+        with Capturing() as output:
+            try:
+                namelists.Namelist.nmls_load(self.mconfig)
+            except SystemExit as e:
+                error = e
+
+        # error needs to exist as the namelist has incorrect syntax
+        assert isinstance(error, SystemExit)
+        for line in output:
```

Comment:
This is just for info while the test runs, right? It can be removed later.

Reply:
Do you mean the for loop with the prints? yes, should I remove it now?

Reply:
done